### PR TITLE
[18.1.x] backport/chore: backport fix; new version chores

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented

back ported #5145 and did new version tasks

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/llmq/blockprocessor.cpp [hunk 1]
```diff
@@ -91,6 +91,19 @@ void CQuorumBlockProcessor::ProcessMessage(CNode* pfrom, const std::string& msg_
             Misbehaving(pfrom->GetId(), 100);
             return;
         }
+        if (pQuorumBaseBlockIndex->nHeight < (::ChainActive().Height() - GetLLMQParams(type).dkgInterval)) {
+            LogPrint(BCLog::LLMQ, "CQuorumBlockProcessor::%s -- block %s is too old, peer=%d\n", __func__,
+                     qc.quorumHash.ToString(), peer.GetId());
```
Comment (UdjinM6) [inline src/llmq/blockprocessor.cpp:96]:
```suggestion
                     qc.quorumHash.ToString(), pfrom->GetId());
```

File: src/llmq/blockprocessor.cpp [hunk 2]
```diff
@@ -91,6 +91,19 @@ void CQuorumBlockProcessor::ProcessMessage(CNode* pfrom, const std::string& msg_
...
+            // TODO: enable punishment in some future version when all/most nodes are running with this fix
+            // Misbehaving(peer.GetId(), 100);
+            return;
+        }
+        if (HasMinedCommitment(type, qc.quorumHash)) {
+            LogPrint(BCLog::LLMQ, "CQuorumBlockProcessor::%s -- commitment for quorum hash[%s], type[%d], quorumIndex[%d] is already mined, peer=%d\n",
+                     __func__, qc.quorumHash.ToString(), uint8_t(type), qc.quorumIndex, peer.GetId());
```
Comment (UdjinM6) [inline src/llmq/blockprocessor.cpp:103]:
```suggestion
                     __func__, qc.quorumHash.ToString(), uint8_t(type), qc.quorumIndex, pfrom->GetId());
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: doc/release-notes.md [hunk 3]
```diff
@@ -1,14 +1,11 @@
-Dash Core version v18.1.0
-=========================
+Dash Core version 18.0.2
```
Comment (thephez) [inline doc/release-notes.md:1]: Shouldn't it be this?
```suggestion
Dash Core version 18.1.1
```

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**